File: Kth Missing Positive Number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int kthMissing=1,curNum=1;
        int arrInd=0;
        while(k>0){
            if(arrInd<arr.size() && arr[arrInd]==curNum)
            arrInd++;
            else{
                k--;
                kthMissing=curNum;
            }
            curNum++;
        }
     return   kthMissing;
            }
};

File: Kth Missing Positive Number.java
class Solution {
    public static int findKthPositive(int[] arr, int k) {
        for(int i=0;i<arr.length;i++)
        {
            int x=arr[i]-(i+1);
            if(x>=k)
                return i+k;
        }
        return arr.length+k;
    }   
  
}
